spec/features/authentication/set_up_two_factor_authentication_spec.rb

```diff
@@ -12,134 +12,92 @@
+    expect(page).to have_content("Set up two factor authentication")
+  end
 
-    it "labels the text key" do
-      expect(page).to have_content("Or, enter this key into your authenticator app:")
-    end
+  it "explains the setup step" do
+    expect(page).to have_content("Scan the QR code")
+  end
 
-    it "presents a text key" do
-      expect(page).to have_content(/(([A-Z0-9]){4}  ){7}[A-Z0-9]{4}/)
-    end
+  it "presents a QR code" do
+    expect(page).to have_css("img[src*='data:image/png;base64']")
+  end
 
-    it "expects a TOTP code" do
-      expect(page).to have_field(:code)
-    end
+  it "labels the text key" do
+    expect(page).to have_content("Or, enter this key into your authenticator app:")
+  end
 
-    it "does not display a link to skip 2fa" do
-      expect(page).to_not have_link("I cannot set up two factor authentication")
-    end
+  it "presents a text key" do
+    expect(page).to have_content(/(([A-Z0-9]){4}  ){7}[A-Z0-9]{4}/)
+  end
 
-    it "rejects directly visiting the unable to two factor authentication page" do
-      visit users_two_factor_authentication_unable_path
-      expect(page).to_not have_current_path(users_two_factor_authentication_unable_path)
-    end
+  it "expects a TOTP code" do
+    expect(page).to have_field(:code)
+  end
 
-    context "when navigating to another page" do
-      before { visit root_path }
+  context "when navigating to another page" do
+    before { visit root_path }
 
-      it "redirects the user back to setup" do
-        expect(page).to have_current_path(users_two_factor_authentication_setup_path)
-      end
+    it "redirects the user back to setup" do
+      expect(page).to have_current_path(users_two_factor_authentication_setup_path)
     end
+  end
 
-    context "when submitting a valid code" do
-      let(:totp_double) { instance_double(ROTP::TOTP) }
+  context "when submitting a valid code" do
+    let(:totp_double) { instance_double(ROTP::TOTP) }
 
-      before do
-        allow(ROTP::TOTP).to receive(:new).and_return(totp_double)
-        allow(totp_double).to receive(:verify).and_return(true)
+    before do
+      allow(ROTP::TOTP).to receive(:new).and_return(totp_double)
+      allow(totp_double).to receive(:verify).and_return(true)
 
-        fill_in :code, with: "999999"
-        click_on "Complete setup"
-      end
+      fill_in :code, with: "999999"
+      click_on "Complete setup"
+    end
 
-      it "authenticates the user" do
-        expect(user.reload.totp_enabled?).to be true
-      end
+    it "authenticates the user" do
+      expect(user.reload.totp_enabled?).to be true
+    end
 
-      it "shows a success message" do
-        expect(page).to have_content("Two factor authentication setup successful")
-      end
+    it "shows a success message" do
+      expect(page).to have_content("Two factor authentication setup successful")
+    end
 
-      it "redirects the user to the admin app" do
+    context "with a super admin user" do
```

Comment:
maybe, now, we can make the default in line 6 a regular user and then set the user here to be a super user? That might make more sense now we don't have any special business logic for the super user anymore.